openroad_interface/place_n_route.py

```diff
@@ -0,0 +1,157 @@
+import re
+import os
+
+import networkx as nx
+
+from .var import directory
+from . import estimation as est  
+from . import detailed as det
+
+def openroad_run():
+    os.chdir(directory)
+    os.system("openroad test.tcl")
+    os.chdir("../../..")
+
+def export_graph(graph, design_name, est_or_det: str):
+    if not os.path.exists("openroad_interface/results/"):
+        os.makedirs("openroad_interface/results/")
+    nx.write_gml(graph, "openroad_interface/results/" + est_or_det + "_" + design_name + ".gml")
+
+def mux_removal(graph: nx.DiGraph, design_name):
+    graph_copy = copy.deepcopy(graph)
+    for node in graph.nodes:
+        if "MUX" in node.upper():
+            input_nodes = list(graph_copy.in_edges(node))
+            output_node = list(graph_copy.out_edges(node))[0][1]
+
+            old_edge_1 = graph_copy[input_nodes[0][0]][node]
+            old_edge_2 = graph_copy[input_nodes[1][0]][node]
+
+            old_output = graph_copy[node][str(output_node)]
+            for attribute in ["net_length", "net_cap", "net_res"]:
+                old_edge_1[attribute] += float(old_output[attribute])
+                old_edge_2[attribute] += float(old_output[attribute])
```

Comment:
I know this is what I originally asked for, but let's instead make it a list of all the Rs and Cs that contributed to this.

Reply:
this well let us generalize our E and T calculation in the next step.